src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt

```diff
@@ -164,4 +172,87 @@ class ExternalPaymentControllerIT(
         // then
         response shouldHaveHttpStatus NOT_FOUND
     }
+
+    should("decide") {
+        // given
+        val decisionRequest = createPaymentDecisionRequest()
+        stubGroupManagerUserGroups(createUserGroupsResponse(GROUP_ID, OTHER_GROUP_ID), OTHER_USER_ID)
+
+        val payment = createPayment()
+        paymentRepository.save(payment)
+
+        // when
+        val response = service.decide(decisionRequest, createGemUser(id = OTHER_USER_ID))
+
+        // then
+        response shouldHaveHttpStatus OK
+    }
+
+    context("return validation exception when decide cause:") {
+        withData(
+            nameFn = { it.first },
+            Pair(PAYMENT_ID_NOT_BLANK, createPaymentDecisionRequest(paymentId = "")),
+            Pair(GROUP_ID_NOT_BLANK, createPaymentDecisionRequest(groupId = "")),
+            Pair(MESSAGE_NULL_OR_NOT_BLANK, createPaymentDecisionRequest(message = "")),
+
+        ) { (expectedMessage, paymentDecisionRequest) ->
+            // when
+            val response = service.decide(paymentDecisionRequest, createGemUser())
+
+            // then
+            response shouldHaveHttpStatus BAD_REQUEST
+            response shouldHaveValidationError expectedMessage
+        }
+    }
+
+    should("return forbidden if user is not a group member") {
```

Comment:
should it be '.. is not a receipt'

Reply:
as told on daily checking if user belongs to the group stays so this test is needed